src/Symfony/Component/Security/Http/Authenticator/AccessToken/AccessTokenAuthenticator.php

```diff
@@ -0,0 +1,156 @@
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Credentials\AccessTokenCredentials;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Throwable;
+
+/**
+ * Provides an implementation of the RFC6750 of an authentication via
+ * an access token.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+class AccessTokenAuthenticator implements AuthenticatorInterface
+{
+    private UserProviderInterface $userProvider;
+    private ?AuthenticationSuccessHandlerInterface $successHandler;
+    private ?AuthenticationFailureHandlerInterface $failureHandler;
+    private AccessTokenHandlerInterface $accessTokenHandler;
+    private ?TranslatorInterface $translator = null;
+    private ?string $realm;
+    /**
+     * @var AccessTokenExtractorInterface[]
+     */
+    private iterable $accessTokenExtractors;
+
+    /**
+     * @param AccessTokenExtractorInterface[] $accessTokenExtractors
+     */
+    public function __construct(UserProviderInterface $userProvider, AccessTokenHandlerInterface $accessTokenHandler, iterable $accessTokenExtractors, AuthenticationSuccessHandlerInterface $successHandler = null, AuthenticationFailureHandlerInterface $failureHandler = null, string $realm = null)
+    {
+        $this->successHandler = $successHandler;
+        $this->failureHandler = $failureHandler;
+        $this->userProvider = $userProvider;
+        $this->accessTokenHandler = $accessTokenHandler;
+        $this->accessTokenExtractors = $accessTokenExtractors;
+        $this->realm = $realm;
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return null !== $this->extractAccessToken($request);
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        $accessToken = $this->extractAccessToken($request);
+        if (null === $accessToken) {
+            throw new BadCredentialsException('Invalid credentials.');
+        }
+        try {
+            $userIdentifier = $this->accessTokenHandler->getUserIdentifierFrom($accessToken);
+        } catch (AuthenticationException $e) {
+            throw $e;
+        } catch (Throwable $e) {
+            throw new BadCredentialsException('Invalid credentials.', 0, $e);
+        }
```

Comment:
```suggestion
        $userIdentifier = $this->accessTokenHandler->getUserIdentifierFrom($accessToken);
```

try/catch blocks like this one hide implementation errors and make the application less debuggable. Unless you have a very good reason to swallow all erros here, we remove that block.